docs/testing_servers.md

```diff
@@ -0,0 +1,340 @@
+# Testing Servers
+
+Testing servers with the conformance runner involves the following steps:
+
+* Implementing the ConformanceService endpoints.
+* Making an executable file that will read messages from `stdin` and write messages to `stdout`.
+* Defining any configuration for what your server supports. For more information on how to do this, see the [Configuration Files](./configuring_and_running_tests.md#configuration-files) section in the [Configuring and Running Tests documentation](./configuring_and_running_tests.md).
+
+## Process
+
+The basic process for working with servers in the conformance suite will work as follows:
+
+* The conformance runner will pass a `ServerCompatRequest` message via `stdin`. This message will contain all the details necessary
+  for your implementation to start its server-under-test. Note that the request does not specify a port to listen on.
+  Implementations should instead pick an available ephemeral port according to their OS and return that value in the response.
+
+  Fields in the request are:
+
+  * `protocol` specifies what protocol will be used to run the tests. 
+  * `http_version` specifies which HTTP version will be used.
```

Comment:
Maybe we could clarify here, too. Some HTTP server libraries may automatically provide support for both HTTP 1.1 and HTTP/2, in which case it is fine for the server program to always support both and ignore this field.

So both of these fields really mean "support _at least_ this protocol and version; fine to also support others".